contracts/evm/contracts/adapters/CentralizedConnection.sol

```diff
@@ -0,0 +1,98 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity >=0.8.0;
+pragma abicoder v2;
+
+import "openzeppelin-contracts-upgradeable/contracts/proxy/utils/Initializable.sol";
+import "@xcall/utils/Types.sol";
+import "@xcall/contracts/xcall/interfaces/IConnection.sol";
+import "@iconfoundation/btp2-solidity-library/interfaces/ICallService.sol";
+
+
+contract XCallCentralizeConnection is Initializable, IConnection {
+
+    mapping(string => uint256) private messageFees;
+    mapping(string => uint256) private responseFees;
+
+    mapping(bytes32 => bool) public seenDeliveryVaaHashes;
```

Comment:
For this you can do it cheaper in both solidity and java by
assert recipts[srcNetwork][sn] == false
recipts[srcNetwork][sn] = true